core/src/main/java/org/infinispan/Cache.java

```diff
@@ -334,6 +335,10 @@
     * concurrently.
     * <p/>
     * If the cache is transactional, it will not interact with the transaction.
+    *
+    * Starting with version 9.3, calling this method does not result if cache entry removed events being fired.
+    * If you rely on cache entry removed events
+    * , use {@link CacheStream#forEach(BiConsumer)} to individually remove each entry and get remove notifications.
```

Comment:
We should document this in the `CacheEntryRemoved` javadoc as well, and check that we don't implicitly rely on the listeners being invoked internally (e.g. `MigrationTask`, global state, L2, JCache/Spring adapters)